### PR TITLE
Revise validated authorities cache lookup

### PR DESCRIPTION
**Pull request by unpluggedk**
#119 

Validated authority look up by UPN doesn't work for ATS with user. 
Changing this to be more direct with an additional parameter that we readily derive before calling the look up.

Also, revise some of the existing unit testing with the naming and the above changes.

### CODE REVIEW

File: MSAL/src/instance/MSALAuthority.m [hunk 1]
```diff
@@ -157,7 +157,9 @@ + (void)resolveEndpointsForAuthority:(NSURL *)unvalidatedAuthority
         tenant = firstPathComponent;
     }
     
-    MSALAuthority *authorityInCache = [MSALAuthority authorityFromCache:updatedAuthority userPrincipalName:userPrincipalName];
+    MSALAuthority *authorityInCache = [MSALAuthority authorityFromCache:updatedAuthority
+                                                          authorityType:authorityType
+                                                      userPrincipalName:userPrincipalName];
```
Comment (oldalton) [inline MSAL/src/instance/MSALAuthority.m]: Should we also respect validate argument here? We're currently doing validation even when validate == NO. Otherwise this argument is unnecessary.

Comment (oldalton) [inline MSAL/src/instance/MSALAuthority.m]: (just in case, I refer to the resolveEndpointsForAuthority:userPrincipalName:validate:context:completionBlock method itself)

File: MSAL/test/unit/MSALAuthorityTests.m [hunk 2]
```diff
@@ -183,34 +183,44 @@ - (void)testIsKnownHost
...
     // Add valid authority
     XCTAssertTrue([MSALAuthority addToValidatedAuthority:aadAuthority userPrincipalName:nil]);
     XCTAssertTrue([MSALAuthority addToValidatedAuthority:b2cAuthority userPrincipalName:nil]);
     XCTAssertTrue([MSALAuthority addToValidatedAuthority:adfsAuthority userPrincipalName:@"user@contoso.com"]);
 
     // Check if valid authority returned
-    MSALAuthority *retrivedAuthority = [MSALAuthority authorityFromCache:aadAuthority.canonicalAuthority userPrincipalName:nil];
+    MSALAuthority *retrivedAuthority = [MSALAuthority authorityFromCache:aadAuthority.canonicalAuthority
+                                                           authorityType:AADAuthority
+                                                       userPrincipalName:nil];
```
Comment (oldalton) [inline MSAL/test/unit/MSALAuthorityTests.m]: It might be a good idea to also add a test case with authorityType == AADAuthority and non-nil userPrincipalName to make sure we test such cases in future too (since it's a valid case for ATS).

File: MSAL/src/instance/MSALAuthority.m (hunk 1 above)
Comment (unpluggedk) [inline MSAL/src/instance/MSALAuthority.m]: Validation portion is in the openIDConfigurationEndpointForAuthority: of the MSAL___AuthorityResolver class. What we always do is discovery portion.

File: MSAL/test/unit/MSALAuthorityTests.m (hunk 2 above)
Comment (unpluggedk) [inline MSAL/test/unit/MSALAuthorityTests.m]: True that! 👍

File: MSAL/src/instance/MSALAuthority.m (hunk 1 above)
Comment (oldalton) [inline MSAL/src/instance/MSALAuthority.m]: true, my bad
